### PR TITLE
Display user-relative date , disable pdf/undo/redo buttons when needed, change some labels

### LINKED ISSUE #171: export PDF action (FE)
- could be disabled on an empty diagram
- should be renamed to PDF, "exportPDF" is too wide
- icon duplicates with on on  "export to JSON"

### LINKED ISSUE #177: link icon to external documentation
It's a good feature but I think its UX should be redesigned (not consistent icon, not visible).

image

### LINKED ISSUE #186: rename "esp-ui-client" to nussknacker-ui

### LINKED ISSUE #187: undo, redo always enabled
Regardless of available redo or undo steps buttons are always enabled.

### LINKED ISSUE #230: processes page
"process name" in table header -> name
  sorting by category -> add secondary sort by name
  last modification -> use relative time (momentjs?)

### PR DESCRIPTION
**Pull request by PPiedel**
This resolves #230
This resolves #187
This resolves #186
This resolves #177
This resolves #171

### CODE REVIEW

File: ui/client/common/DateUtils.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
 
 export default {
-
-
-  format(dateTimeString) {
-    //TODO: something better?
-    return dateTimeString.replace("T", " | ").substring(0, 18)
-  }
-
+    format(dateTimeString) {
```
Comment (mproch) [inline ui/client/common/DateUtils.js]: do we really want this? I think our users will be a bit confused. There *must* be at least some way of showing absolute date...

File: ui/client/common/ProcessUtils.js [hunk 2]
```diff
@@ -9,6 +9,11 @@ class ProcessUtils {
     return !_.isEmpty(fetchedProcessDetails) ? _.isEqual(fetchedProcessDetails.json, processToDisplay) : true
   }
 
+  nothingToExport = (state) => {
```
Comment (lciolecki) [inline ui/client/common/ProcessUtils.js]: Can we named this method better? Something like availableToExport or just canExport?

File: ui/client/components/right-panel/UserRightPanel.js [hunk 3]
```diff
@@ -133,8 +133,8 @@ class UserRightPanel extends Component {
         {name: "migrate", visible: this.props.capabilities.deploy && !_.isEmpty(this.props.featuresSettings.remoteEnvironment), disabled: !conf.deployPossible, onClick: this.migrate, icon: InlinedSvgs.buttonMigrate},
         {name: "compare", onClick: this.compareVersions, icon: 'compare.svg', disabled: this.hasOneVersion()},
         {name: "import", visible: this.props.capabilities.write, disabled: false, onClick: this.importProcess, icon: InlinedSvgs.buttonImport, dropzone: true},
-        {name: "export", onClick: this.exportProcess, icon: InlinedSvgs.buttonExport},
-        {name: "exportPDF", disabled: !this.props.nothingToSave, onClick: this.exportProcessToPdf, icon: InlinedSvgs.buttonExport},
+        {name: "export", disabled: this.props.nothingToExport, onClick: this.exportProcess, icon: InlinedSvgs.buttonExport},
+        {name: "PDF", disabled: this.props.nothingToExport, onClick: this.exportProcessToPdf, icon: InlinedSvgs.pdf},
```
Comment (lciolecki) [inline ui/client/components/right-panel/UserRightPanel.js]: Export PDF?

File: ui/client/common/DateUtils.js (hunk 1 above)
Comment (PPiedel) [reply ui/client/common/DateUtils.js]: What would you say if the absolute date would be displayed "on mouse over" specified date?

File: ui/client/components/right-panel/UserRightPanel.js (hunk 3 above)
Comment (PPiedel) [reply ui/client/components/right-panel/UserRightPanel.js]: Sorry, I forgot to include link to this issue: https://github.com/TouK/nussknacker/issues/171

Comment (lciolecki) [reply ui/client/components/right-panel/UserRightPanel.js]: Okay, so leave it as PDF, but maybe in export we can put JSON?

Comment (lciolecki) [reply ui/client/components/right-panel/UserRightPanel.js]: Oh, I have an idea: at title put full description like: Export to PDF file. And at rest of buttons you can add similar full description.

Comment (PPiedel) [reply ui/client/components/right-panel/UserRightPanel.js]: Good idea, will apply that

File: ui/client/common/DateUtils.js (hunk 1 above)
Comment (mproch) [reply ui/client/common/DateUtils.js]: maybe we can try it...

**Pull request merged**